File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/TaskFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.ConversionProcess;
import com.rapyutarobotics.alica.Tags;
import de.unikassel.vs.alica.planDesigner.alicamodel.Task;
import org.w3c.dom.Element;

public class TaskFactory extends Factory {

    public static Task create (Element taskNode, ConversionProcess cp) {
        Task task = new Task();
        Factory.setAttributes(taskNode, task);
        // the description field is obsolete -> use comments field
        if (task.getComment().isEmpty()) {
            task.setComment(taskNode.getAttribute(Tags.DESCRIPTION));
        }
        cp.addElement(task);
        return task;
    }
}

File: alica_engine/src/engine/modelmanagement/factories/TaskFactory.cpp
#include "engine/modelmanagement/factories/TaskFactory.h"
#include "engine/model/Task.h"
#include "engine/modelmanagement/Strings.h"

namespace alica
{
Task* TaskFactory::create(const YAML::Node& taskNode, TaskRepository* taskRepository)
{
    Task* task = new Task(taskRepository);
    Factory::setAttributes(taskNode, task);
    Factory::storeElement(task, alica::Strings::task);
    return task;
}
} // namespace alica
